lib/reverse_sentence.rb

```diff
@@ -1,4 +1,41 @@
 # A method to reverse the words in a sentence, in place.
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
+def reverse_sentence(my_string)
+  return nil if my_string.nil?
+      half = my_string.length / 2
+      j = my_string.length - 1
+      temp_variable = " "
+      i = 0
+
+      while i < half
```

Comment:
Consider creating a method that takes in the string, the starting index and ending index which reverses the characters in the string between starting index and ending index. Then you can call that method from this method and avoid repeated code.